### PR TITLE
Test unmarshalling CSRF token

### LINKED ISSUE #207: CSRFTokenCredentialRequestAuth can not be unmarshaled properly
**Description**

Unmarshaling and API which contains the CSRF token do not work. The issue is with `Credential` field of type `CredentialData`. The interface needs a custom implementation of `UnmarshalJSON` method, as it is implemented for the `Auth` struct [here](https://github.com/kyma-incubator/compass/blob/master/components/director/pkg/graphql/auth.go).

**AC**:
- Write `UnmarshalJSON` for `CSRFTokenCredentialRequestAuth`
- Write integration test for this case

**Expected result**

The `CSRFTokenCredentialRequestAuth` is marshaled properly.

**Actual result**

Unmarshalling APIDefinition with CSRF returns an error: `decoding response: json: cannot unmarshal object into Go struct field APIDefinition.defaultAuth of type graphql.CredentialData`

### PR DESCRIPTION
**Pull request by dbadura**
**Description**

Changes proposed in this pull request:

-modified e2e tests, to check if `CSRFTokenCredentialRequestAuth` is unmarshaled properly

**Related issue(s)**

Closes #207

### CODE REVIEW

File: tests/end-to-end/director/fixtures.go [hunk 1]
```diff
@@ -37,6 +32,23 @@ func fixOauthAuth() *graphql.AuthInput {
...
 
+func fixBasicCredential() *graphql.CredentialDataInput {
+	return &graphql.CredentialDataInput{
+		Basic: &graphql.BasicCredentialDataInput{
+			Username: "admin",
+			Password: "secret",
+		}}
+}
+
+func fixOauthCredential() *graphql.CredentialDataInput {
```
Comment (pkosiec) [inline tests/end-to-end/director/fixtures.go]: `OAuth`

File: tests/end-to-end/director/fixtures.go [hunk 2]
```diff
@@ -37,6 +32,23 @@ func fixOauthAuth() *graphql.AuthInput {
...
+	return &graphql.CredentialDataInput{
+		Oauth: &graphql.OAuthCredentialDataInput{
+			URL:          "url.net",
+			ClientSecret: "grazynasectet",
```
Comment (pkosiec) [inline tests/end-to-end/director/fixtures.go]: `grazynasecret`

File: tests/end-to-end/director/graphqlizer_test.go [hunk 3]
```diff
@@ -86,31 +85,56 @@ func (g *graphqlizer) FetchRequestInputToGQL(in *graphql.FetchRequestInput) (str
...
+			{{- if .Basic }}
 			basic: {
-				username: "{{ .Credential.Basic.Username}}",
-				password: "{{ .Credential.Basic.Password}}",
+				username: "{{ .Basic.Username }}",
+				password: "{{ .Basic.Password }}",
 			}
 			{{- end }}
-			{{- if .Credential.Oauth }}
+			{{- if .Oauth }}
```
Comment (pkosiec) [inline tests/end-to-end/director/graphqlizer_test.go]: `.OAuth`

File: tests/end-to-end/director/graphqlizer_test.go [hunk 4]
```diff
@@ -86,31 +85,56 @@ func (g *graphqlizer) FetchRequestInputToGQL(in *graphql.FetchRequestInput) (str
...
 			oauth: {
-				clientId: "{{ .Credential.Oauth.ClientID}}",
-				clientSecret: "{{ .Credential.Oauth.ClientSecret }}",
-				url: "{{ .Credential.Oauth.URL }}",
+				clientId: "{{ .Oauth.ClientID }}",
+				clientSecret: "{{ .Oauth.ClientSecret }}",
+				url: "{{ .Oauth.URL }}",
```
Comment (pkosiec) [inline tests/end-to-end/director/graphqlizer_test.go]: `OAuth`

File: tests/end-to-end/director/graphqlizer_test.go [hunk 5]
```diff
@@ -86,31 +85,56 @@ func (g *graphqlizer) FetchRequestInputToGQL(in *graphql.FetchRequestInput) (str
...
 	return g.genericToGQL(in, `{
-		credential:{
-			{{- if .Credential.Basic }}
+		{{- if .Csrf }}
+		csrf: {{ CSRFTokenCredentialRequestAuthInputToGQL .Csrf }}
+		{{- end }}
+	}`)
+}
+
+func (g *graphqlizer) CredentialToGQL(in *graphql.CredentialDataInput) (string, error) {
```
Comment (tgorgol) [inline tests/end-to-end/director/graphqlizer_test.go]: Shouldn't it be named `CredentialDataInputToGQL`?

**Pull request merged**